### PR TITLE
[RCF-334] Document type dropdowns should be display in selected languages

### PR DESCRIPTION
**Pull request by SachinPremkumar**
[RCF-334]

[RCF-334]: https://mosip.atlassian.net/browse/RCF-334?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/dao/ApplicantValidDocumentDao.java [hunk 1]
```diff
@@ -17,6 +17,9 @@ public interface ApplicantValidDocumentDao {
     @Query("select distinct doc_type_code from applicant_valid_doc where doc_cat_code=:docCategoryCode")
     List<String> findAllDocTypesByDocCategory(String docCategoryCode);
 
+    @Query("select name from document_type where code=:code and lang_code=:langCode")
+    List<String> getDBValue(String code,String langCode);
```
Comment (jainhitesh9998) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/dao/ApplicantValidDocumentDao.java:21]: can we change the name of method, as it doesn't reflect the action

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java [hunk 2]
```diff
@@ -18,11 +20,18 @@ public ApplicantValidDocRepository(ApplicantValidDocumentDao applicantValidDocum
...
+        List<String> nameList;
+        List<String> documentList = new ArrayList<>();
+        if (applicantType == null) {
+            nameList = this.applicantValidDocumentDao.findAllDocTypesByDocCategory(categoryCode);
+        }else {
+            nameList = this.applicantValidDocumentDao.findAllDocTypesByDocCategoryAndApplicantType(applicantType,
+                    categoryCode);
+        }
+        nameList.forEach((v) -> {
+            documentList.add(this.applicantValidDocumentDao.getDBValue(v,langCode).get(0));
```
Comment (jainhitesh9998) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java:32]: this maybe prone to nullpointer exception, we need to handle such cases

**Review by jainhitesh9998 — CHANGES_REQUESTED** [review]

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/dao/ApplicantValidDocumentDao.java (hunk 1 above)
Comment (SachinPremkumar) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/dao/ApplicantValidDocumentDao.java]: modified the method name

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java (hunk 2 above)
Comment (SachinPremkumar) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java]: handle the null pointer exception

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java [hunk 3]
```diff
@@ -18,11 +20,20 @@ public ApplicantValidDocRepository(ApplicantValidDocumentDao applicantValidDocum
...
+        List<String> documentList = new ArrayList<>();
+        if (applicantType == null) {
+            nameList = this.applicantValidDocumentDao.findAllDocTypesByDocCategory(categoryCode);
+        }else {
+            nameList = this.applicantValidDocumentDao.findAllDocTypesByDocCategoryAndApplicantType(applicantType,
+                    categoryCode);
+        }
+        nameList.forEach((v) -> {
+            if(v!=null) {
+                documentList.add(this.applicantValidDocumentDao.findAllDocTypesByLanguageCode(v, langCode).get(0));
```
Comment (jainhitesh9998) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java:33]: .get(0) can be null

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java [hunk 4]
```diff
@@ -18,11 +20,20 @@ public ApplicantValidDocRepository(ApplicantValidDocumentDao applicantValidDocum
     }
 
     public List<String> getDocumentTypes(String applicantType, String categoryCode, String langCode) {
-        if (applicantType == null)
-            return this.applicantValidDocumentDao.findAllDocTypesByDocCategory(categoryCode);
-
-        return this.applicantValidDocumentDao.findAllDocTypesByDocCategoryAndApplicantType(applicantType,
-                categoryCode);
+        List<String> nameList;
```
Comment (jainhitesh9998) [inline android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java:23]: let's rename it to docTypeList

**Review by jainhitesh9998 — CHANGES_REQUESTED** [review]

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java (hunk 4 above)
Comment (SachinPremkumar) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java]: changed the variable name

File: android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java (hunk 3 above)
Comment (SachinPremkumar) [reply android/clientmanager/src/main/java/io/mosip/registration/clientmanager/repository/ApplicantValidDocRepository.java]: added the null handler

**Review by jainhitesh9998 — APPROVED** [review]

**Pull request merged**